src/fsharp/TypeChecker.fs

```diff
@@ -8271,6 +8271,17 @@ and TcFunctionApplicationThen cenv overallTy env tpenv mExprAndArg expr exprty (
         | SynExpr.CompExpr (false,_isNotNakedRefCell,comp,_m) -> 
             let bodyOfCompExpr,tpenv = TcComputationOrSequenceExpression cenv env overallTy mFunExpr (Some(expr.Expr,exprty)) tpenv comp
             TcDelayed cenv overallTy env tpenv mExprAndArg (MakeApplicableExprNoFlex cenv bodyOfCompExpr) (tyOfExpr cenv.g bodyOfCompExpr) ExprAtomicFlag.NonAtomic delayed 
+        | SynExpr.LongIdent(_, LongIdentWithDots(idents, _), _, _) when not (isNil idents) ->
+            match expr with
+            // `nameof` operator
+            | ApplicableExpr (_, Expr.App(Expr.Const(Const.String("nameof"), _, _), _, _, _, _), _) -> // no idea really what shape we should match on here, will check in debugger
```

Comment:
this is technically a breaking change when user-defined function `nameof` exists in the scope.  I think it should resolve the name and only treat it as `nameof` operator if it it was resolved to intrinsic from FSharp.Core